### PR TITLE
RND-313 Add `LoginPage` component along with its dependencies

### PR DESCRIPTION
**Pull request by kubama**
## Description
Added `LoginPage` component along with its dependencies. It's going to be reused in composer.

## Screenshots / Videos
All components are extracted from stage and preserve their original look'n'feel

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
Added stories, snapshots, loki images.
Added unit test for `LoginPage`.

 

## Documentation
Added JSDocs

### CODE REVIEW

**Comment by PatrycjaKubiczek** [comment]
@kubama there is a lot of changes so I won't make it with review today, I'll do it on Monday morning

File: src/components/cloudify/LoginPage/LoginPage.tsx [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+     */
+    contextPath: string;
+
+    /**
+     * Path assigned to the page which this component os supposed to be used with
+     */
+    defaultLoginPagePath: string;
+
+    /**
+     * Render even handler
```
Comment (PatrycjaKubiczek) [inline src/components/cloudify/LoginPage/LoginPage.tsx:50]:
```suggestion
     * Render event handler
```

File: src/components/cloudify/LoginPage/LoginPage.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+     */
+    onRender?: () => void;
+
+    /**
+     * Wheter to enable first login hint
```
Comment (PatrycjaKubiczek) [inline src/components/cloudify/LoginPage/LoginPage.tsx:55]:
```suggestion
     * Whether to enable first login hint
```

File: src/components/elements/LogoLabel/LogoLabel.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export default {
+    title: 'Elements/LogoLabel',
+    component: LogoLabel,
+    decorators: [LiveEditDecorator({ LogoLabel })]
+};
+
+type LogoLabelStory = ComponentStory<typeof LogoLabel>;
+
+export const basic: LogoLabelStory = () => <LogoLabel color="black" content="This is a logo label" />;
+basic.storyName = 'Default';
```
Comment (PatrycjaKubiczek) [inline src/components/elements/LogoLabel/LogoLabel.stories.tsx:16]: there is no need to add `basic.storyName = 'Default';` it won't display - I'm currently working on dedicated [ticket ](https://cloudifysource.atlassian.net/browse/RD-6339)to remove this from components repo

File: src/components/layout/LogoPage/LogoPage.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+type LogoLabelStory = ComponentStory<typeof LogoPage>;
+
+export const basic: LogoLabelStory = () => (
+    <div style={{ height: 300 }}>
+        <LogoPage>Status text...</LogoPage>
+    </div>
+);
+basic.storyName = 'Default';
```
Comment (PatrycjaKubiczek) [inline src/components/layout/LogoPage/LogoPage.stories.tsx:20]: there is no need to add basic.storyName = 'Default'; it won't display - I'm currently working on dedicated [ticket ](https://cloudifysource.atlassian.net/browse/RD-6339)to remove this from components repo

File: src/components/cloudify/LargeProductLogo/LargeProductLogo.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+};
+
+type LargeProductLogoStory = ComponentStory<typeof LargeProductLogo>;
+
+export const basic: LargeProductLogoStory = () => (
+    <div style={{ background: 'blue' }}>
+        <LargeProductLogo />
+    </div>
+);
+basic.storyName = 'Default';
```
Comment (PatrycjaKubiczek) [inline src/components/cloudify/LargeProductLogo/LargeProductLogo.stories.tsx:20]: there is no need to add basic.storyName = 'Default'; it won't display - I'm currently working on dedicated [ticket ](https://cloudifysource.atlassian.net/browse/RD-6339)to remove this from components repo

File: src/components/cloudify/LoginPage/LoginPage.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+                    password: 'Password',
+                    username: 'Username',
+                    submit: 'LOGIN',
+                    firstLoginHint: ''
+                }[translationKey])
+            }
+        />
+    </div>
+);
+basic.storyName = 'Default';
```
Comment (PatrycjaKubiczek) [inline src/components/cloudify/LoginPage/LoginPage.stories.tsx:40]: there is no need to add basic.storyName = 'Default'; it won't display - I'm currently working on dedicated [ticket ](https://cloudifysource.atlassian.net/browse/RD-6339)to remove this from components repo

File: src/components/cloudify/LoginPage/LoginPage.tsx [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+        display: block !important;
+        margin: 30px auto !important;
+        min-width: 50% !important;
+        border-radius: 25px !important;
+        color: ${colors.white} !important;
+    }
+`;
+
+/**
+ * Content of the generic product logic page.
```
Comment (PatrycjaKubiczek) [inline src/components/cloudify/LoginPage/LoginPage.tsx:142]: should it be logic or login? 🤔

**Review by PatrycjaKubiczek — CHANGES_REQUESTED** [review]
Sorry for late review, I was a bit occupied with my own PR ;) 

Looks really good, noticed a couple of minor issues. Besides the things with comments there is one more:
https://github.com/cloudify-cosmo/cloudify-ui-components/blob/ff4aed4f303230d9ec5ed90bda7d3c6893ef6690/src/components/elements/Logo/Logo.tsx#L24 I also notice there is typo in Logo.tsx in `45px x 45x` it should be `45px x 45px`

File: src/components/elements/LogoLabel/LogoLabel.stories.tsx (hunk 3 above)
Comment (kubama) [reply src/components/elements/LogoLabel/LogoLabel.stories.tsx]: Fixed

File: src/components/cloudify/LargeProductLogo/LargeProductLogo.stories.tsx (hunk 5 above)
Comment (kubama) [reply src/components/cloudify/LargeProductLogo/LargeProductLogo.stories.tsx]: Fixed

File: src/components/cloudify/LoginPage/LoginPage.stories.tsx (hunk 6 above)
Comment (kubama) [reply src/components/cloudify/LoginPage/LoginPage.stories.tsx]: Fixed

**Comment by kubama** [comment]
> I also notice there is typo in Logo.tsx in `45px x 45x` it should be `45px x 45px`

Fixed

File: src/components/cloudify/LoginPage/LoginPage.tsx [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+                                        value={password}
+                                        onChange={handleInputChange}
+                                    />
+                                </Form.Field>
+                            </div>
+                        }
+                    />
+
+                    {error && (
+                        <Message error style={{ display: 'block', backgroundColor: '#fdeded', boxShadow: 'none' }}>
```
[the remark this answers is not in the document]
Comment (kubama) [reply src/components/cloudify/LoginPage/LoginPage.tsx:270]: I searched through the project files and didn't find it defined elsewhere

File: src/components/cloudify/LoginPage/LoginPage.tsx [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+
+    /**
+     * Text color
+     */
+    loginPageTextColor: string;
+
+    /**
+     * Header color
+     */
+    loginPageHeaderColor: string;
```
[the remark this answers is not in the document]
Comment (kubama) [reply src/components/cloudify/LoginPage/LoginPage.tsx:72]: This component is to be reused in stage as well, so it needs to preserve all props that are needed there

[the remark this answers is not in the document]
Comment (qooban) [reply src/components/cloudify/LoginPage/LoginPage.tsx:72]: I know. I thought we already use there theme context in Stage for this component, but I see right now we have there `whiteLabel` prop.

In that case let it be like that.

**Review by qooban — APPROVED** [review]

**Review by PatrycjaKubiczek — APPROVED** [review]

**Pull request merged**